oidc-exchange.py

```diff
@@ -0,0 +1,154 @@
+import os
+import sys
+from http import HTTPStatus
+from pathlib import Path
+from typing import NoReturn
+from urllib.parse import urlparse
+
+import id  # pylint: disable=redefined-builtin
+import requests
+
+_GITHUB_STEP_SUMMARY = Path(os.getenv("GITHUB_STEP_SUMMARY"))
+
+# Rendered if OIDC identity token retrieval fails for any reason.
+_TOKEN_RETRIEVAL_FAILED_MESSAGE = """
+OIDC token retrieval failed: {identity_error}
+
+This generally indicates a workflow configuration error, such as insufficient
+permissions. Make sure that your workflow has `id-token: write` configured
+at either the workflow or job level, e.g.:
```

Comment:
Let's not give people ideas. OIDC privilege gives the workflow access not only to PyPI, but potentially to any number of other external systems too. Giving all workflow jobs this privilege is certainly a conscious choice to be made, not something to copy from an example just because the guide mentioned it.
```suggestion
at the job level, e.g.:
```